### PR TITLE
Wordle

### PR DESCRIPTION
**Pull request by MrNesli**
Issue #69 
Adds a **wordle game** to the bot. Probably improved version of this: https://top.gg/bot/939625136900960286.

**New commands:**
``` /wordle play ``` - starts the game
``` /wordle leaderboard top=3``` - sends an embed with top 3 players of the wordle game

### CODE REVIEW

File: bot/extensions/wordle/wordle_game.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+from random import choice as random_choice
+
+
+class WordleGuess(Enum):
+    GOOD = 1
+    PARTIALLY = 2
+    WRONG = 3
+    EMPTY = 4
+
+WORDLE_PROCESSED_DICT = Dict[Tuple[str, int], WordleGuess]
```
[the remark this answers is not in the document]
Comment (PenguinBoi12) [reply bot/extensions/wordle/wordle_game.py:12]: If it's supposed to be some kind of type, it shouldn't be all upcased but rather use the same notation as classes. So `WORDLE_PROCESSED_DICT` might need to be `WordleProcesses` (I would also remove the `_DICT`). It also needs to be a [TypeVar](https://dev.to/decorator_factory/typevars-explained-hmo). 

Again, that is if it's supposed to be a type which seems like it.

File: bot/extensions/wordle/wordle_game.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+            :returns: True if the guess is complete, otherwise False
+
+            :rtype: bool
+        """
+        if len(self.__guess) == self.WORDLE_LENGTH:
+            return True
+        return False
+
+    @staticmethod
```
Comment (PenguinBoi12) [inline bot/extensions/wordle/wordle_game.py:50]: Why is it static? Isn't WordleGame a game for each user?

File: bot/models/extensions/games/wordle.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+class Wordle(app.base, Model):
+	__tablename__ = 'wordle'
+
+	id = Column(Integer, primary_key=True)
+	member_id = Column(Integer, nullable=False, unique=True)
+	points = Column(Integer, nullable=False)
+	play_date = Column(DateTime, nullable=False)
+
+	@classmethod
+	def update_database(cls, member_id: int, points: int):
```
Comment (PenguinBoi12) [inline bot/models/extensions/games/wordle.py:16]: As I said in the other PR, that's not how you use models. I think you should be able to adjust that from what we talked about. If not, you can ask.

File: bot/models/extensions/games/wordle.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+from sqlalchemy import Column, Integer, DateTime
+from bot import app
+from db.model import Model
+from datetime import datetime
+
+
+class Wordle(app.base, Model):
+	__tablename__ = 'wordle'
```
Comment (PenguinBoi12) [inline bot/models/extensions/games/wordle.py:8]: `wordles`

**Review by PenguinBoi12 — CHANGES_REQUESTED** [review]
There's a lot of conflicts that need to be solved and it seems like you changed some things that you shouldn't have change.

We should show the rules of the game here. Also, fix the color of the embed.
image

Put all the assets in one folder called wordle. (`bot/assets/wordle`)

Interesting... You need to test more cases.
image

I feel like in general you should try to restructure everything based on what we talked about in the voting PR.

File: bot/models/extensions/games/wordle.py (hunk 3 above)
Comment (MrNesli) [reply bot/models/extensions/games/wordle.py:16]: I mean I created this function just for convenience, idk

File: bot/extensions/wordle/wordle_game.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrNesli) [reply bot/extensions/wordle/wordle_game.py:12]: pep :anger:

**Pull request closed without merging**